### PR TITLE
[v6r18] Fixing missing socket object for PlainTransport

### PR DESCRIPTION
**Pull request by FlorentNKada**

### CODE REVIEW

File: Core/DISET/private/Transports/BaseTransport.py [hunk 1]
```diff
@@ -36,7 +38,7 @@ def __init__( self, stServerAddress, bServerMode = False, **kwargs ):
     self.sentKeepAlives = 0
     self.waitingForKeepAlivePong = False
     self.__keepAliveLapse = 0
-    self.oSocket = None
+    self.oSocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```
Comment (atsareg) [inline Core/DISET/private/Transports/BaseTransport.py]: Why do you need this ? The self.oSocket will anyway be recreated in initAsXXX() functions

Comment (fstagni) [inline Core/DISET/private/Transports/BaseTransport.py]: This specific line is probably not needed indeed, but without the proposed changes in PlainTransport there's no socket created at all. 
Which means that BTW as of now without this PR the "dip" protocol is not usable.

Comment (atsareg) [inline Core/DISET/private/Transports/BaseTransport.py]: Yes, I agree. I would only just keep the fix in the PlainTransport.py

Comment (fstagni) [inline Core/DISET/private/Transports/BaseTransport.py]: @FlorentNKada can you please update your PR with what @atsareg suggests?

File: Core/DISET/private/Transports/PlainTransport.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ def initAsClient( self ):
     if 'timeout' in self.extraArgsDict:
       timeout = self.extraArgsDict[ 'timeout' ]
     try:
-      self.oSocket.create_connection( self.stServerAddress, timeout )
+      self.oSocket = socket.create_connection(self.stServerAddress, timeout)
```
Comment (chaen) [inline Core/DISET/private/Transports/PlainTransport.py]: @FlorentNKada  didn't you also need to fix ```checkSanity``` ?

Comment (FlorentNKada) [inline Core/DISET/private/Transports/PlainTransport.py]: I have tried to initialize the socket in PlainTransport.py, but it doesn't seem to work. It raises errors in BaseClient.py and RPCClient.py. I am working on it.

Do we need to fix checkSanity ? I didn't really get what this function actually does. It seems like it just returns S_OK().

File: Core/DISET/private/Transports/PlainTransport.py [hunk 3]
```diff
@@ -15,7 +15,7 @@ def initAsClient( self ):
     if 'timeout' in self.extraArgsDict:
       timeout = self.extraArgsDict[ 'timeout' ]
     try:
-      self.oSocket.create_connection( self.stServerAddress, timeout )
+        self.oSocket = socket.create_connection(self.stServerAddress, timeout)
```
Comment (atsareg) [inline Core/DISET/private/Transports/PlainTransport.py]: According to DIRAC coding conventions, the identation is 2 spaces

File: Core/DISET/private/Transports/PlainTransport.py [hunk 4]
```diff
@@ -114,7 +114,7 @@ def _write( self, buffer ):
     return S_OK( sentBytes )
 
 def checkSanity( *args, **kwargs ):
-  return S_OK()
+  return S_OK({})
```
Comment (atsareg) [inline Core/DISET/private/Transports/PlainTransport.py]: According to DIRAC coding conventions this should be:

   return S_OK( {} )

Comment (chaen) [inline Core/DISET/private/Transports/PlainTransport.py]: Not according to pep8, which we agreed to accept ;-)

**Pull request merged**